### PR TITLE
Make STRAC data feed work for all states

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
This PR fixes a bug where a report from STRAC would fail on many states. 

## Changes
- Added mappings to the STRAC schema to provide required data elements
- Moved a mapping for PA from the STRAC schema to PA schema

## Checklist
- [x] Tested locally?
- [x] Ran `quickTest all`?
- [x] Ran `./prime test` against local Docker ReportStream container?
- [x] Downloaded a file from http://localhost:7071/api/download
- [ ] Updated the release notes? 
- [ ] Added tests?
- [ ] Did you check for sensitive data, and remove any? 
- [ ] Does logging contain sensitive data?  
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Other testing
Generated fake STRAC data for AZ. This data failed before because of missing elements. Now it succeeds.
```
rickhawes@MacBook-Pro-16 ~/P/p/prime-router (rick/fix-strac-to-allstates)> ./prime data --input result_files/fake-az-strac.csv --input-schema=strac/strac-covid-19 --route --output-dir routed_files
10:37:19.263 [main][1] INFO  ca.uhn.hl7v2.util.Home - hapi.home is set to /Users/rickhawes/Projects/prime-data-hub/prime-router/.
10:37:19.274 [main][1] INFO  ca.uhn.hl7v2.VersionLogger - HAPI version is: 2.3
10:37:19.277 [main][1] INFO  ca.uhn.hl7v2.VersionLogger - Default Structure libraries found for HL7 versions 2.5.1,
Loaded schema and receivers
Opened: /Users/rickhawes/Projects/prime-data-hub/prime-router/result_files/fake-az-strac.csv
Creating these files:
/Users/rickhawes/Projects/prime-data-hub/prime-router/routed_files/az-covid-19-917da57a-3eed-4e28-9ef2-bd02a91700a0-20210423103719.csv
/Users/rickhawes/Projects/prime-data-hub/prime-router/routed_files/covid-19-295a5bf6-076a-4e8e-a0d9-3a2190028c34-20210423103719.hl7
/Users/rickhawes/Projects/prime-data-hub/prime-router/routed_files/pima-az-covid-19-ef4184a1-a1dc-48f8-aa9d-5686848e86c3-20210423103719.csv
rickhawes@MacBook-Pro-16 ~/P/p/prime-router (rick/fix-strac-to-allstates)>
```

## Security
None

## Fixes
*List GitHub issues this PR fixes*
- #632

## To Be Done
None

### CODE REVIEW

File: prime-router/metadata/schemas/PA/pa-covid-19-redox.schema [hunk 1]
```diff
@@ -32,6 +32,10 @@ elements:
     mapper: lookup(ordering_provider_state, ordering_provider_county)
     redoxOutputFields: ["Orders[0].Provider.Address.County"]
 
+  # Yes, PA wants the facility phone number mapped to provider
+  - name: ordering_provider_phone_number
+    mapper: use(ordering_facility_phone_number)
+
   - name: ordering_facility_county
```
Comment (RickHawesUSDS) [inline prime-router/metadata/schemas/PA/pa-covid-19-redox.schema:39]: This mapping was in the STRAC schema.

File: prime-router/metadata/schemas/STRAC/strac-covid-19.schema [hunk 2]
```diff
@@ -18,8 +18,7 @@ elements:
   - name: ordering_facility_name
     csvFields: [{name: Ordering_Facility_Name}]
 
-  # Yes, PA wants the facility phone number mapped to provider
-  - name: ordering_provider_phone_number
+  - name: ordering_facility_phone_number
     csvFields: [{name: Ordering_Facility_Phone_Number}]
```
Comment (RickHawesUSDS) [inline prime-router/metadata/schemas/STRAC/strac-covid-19.schema:22]: Making `Ordering_Facility_Phone_Number` map to `ordering_facility_phone_number` as you would expect. Now the PA hack is done in the PA schema.

File: prime-router/metadata/schemas/STRAC/strac-covid-19.schema [hunk 3]
```diff
@@ -167,3 +166,11 @@ elements:
 
   - name: patient_id_type
     mapper: ifPresent(patient_id, PI)
+
+  # the kit_id is the serial number on the test kit
+  - name: message_id
+    mapper: use(filler_order_id)
+
```
Comment (RickHawesUSDS) [inline prime-router/metadata/schemas/STRAC/strac-covid-19.schema:173]: The `message_id` is a required.

File: prime-router/src/main/kotlin/cli/TestReportStream.kt [hunk 4]
```diff
@@ -704,13 +704,14 @@ class Strac : CoolTest() {
             val tree = jacksonObjectMapper().readTree(json)
             val reportId = ReportId.fromString(tree["id"].textValue())
             echo("Id of submitted report: $reportId")
+            val expectedWarningCount = 0
             val warningCount = tree["warningCount"].intValue()
-            if (warningCount == allGoodReceivers.size - 1) {
+            if (warningCount == expectedWarningCount) {
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/cli/TestReportStream.kt:709]: :-)

**Review by jimduff-usds — APPROVED** [review]
Your timing is excellent.   See Mo's #922  which would generate lots of Warnings for poor Strac if your fix here was not in place.    I'd like to get both that change and this one into Staging, then do some thorough all-50-states testing, sending both Strac and SimpleReport data all over the place.

Nice!

**Pull request merged**